packages/blade/src/components/Typography/Text/Text.tsx

```diff
@@ -0,0 +1,97 @@
+import type { ReactElement } from 'react';
+import styled from 'styled-components';
+import type { TextTypes } from '../../../tokens/theme/theme';
+import getPlatform from '../../../utils/getPlatform';
+import type { Theme } from '../../BladeProvider';
+import BaseText from '../BaseText';
+import type { BaseTextProps } from '../BaseText/BaseText';
+
+type TextCommonProps = {
+  type: TextTypes;
+  truncateAfterLines?: number;
+  children?: React.ReactNode;
+};
+
+type TextBodyVariant = TextCommonProps & {
+  variant: 'body';
+  weight: keyof Theme['typography']['fonts']['weight'];
+};
+
+type TextCaptionVariant = TextCommonProps & {
+  variant: 'caption';
+  weight: keyof Pick<Theme['typography']['fonts']['weight'], 'regular'>;
```

Comment:
Evaluated type here is a [this](https://github.com/razorpay/blade/blob/57635d733d4d0635a2a6072c55629d6f3e560d3d/packages/blade/src/tokens/global/typography.ts#L5)? And above in `TextBodyVariant` is an [object](https://github.com/razorpay/blade/blob/57635d733d4d0635a2a6072c55629d6f3e560d3d/packages/blade/src/tokens/global/typography.ts#L4-L7)?
Seems like inconsistent for the same prop name `weight`.

Reply:
lol this looks soo overkill because there's just two values 'regular' & 'bold' :p

Reply:
consistency 🙈